File: go/mylife-energy/pkg/api/stats.go
package api

import (
	"fmt"
	"mylife-energy/pkg/business/stats"
	"mylife-tools-server/services/api"
	"mylife-tools-server/services/sessions"
	"time"
)

// jour -> toutes les 15 mins
// mois -> tous les jours
// annee -> tous les mois

var statsDef = api.MakeDefinition("stats", getValues)

type statsInput struct {
	Type      stats.StatsType
	Timestamp time.Time
	Sensors   []string
}

type statsOutput = stats.StatValues

func getValues(session *sessions.Session, arg statsInput) ([]statsOutput, error) {
	a, b := stats.GetValues(arg.Type, arg.Timestamp, arg.Sensors)
	fmt.Printf("+%v +%v\n", a, b)
	return a, b
}

File: mylife-energy/ui/src/api/services/stats.ts
import { api } from 'mylife-tools';
import { StatsType, StatValue } from '..';

export class Stats extends api.services.Service {
  async getValues({ type, timestamp, sensors }: { type: StatsType; timestamp: Date; sensors: string[] }) {
    return (await this.call({
      service: 'stats',
      method: 'getValues',
      type,
      timestamp,
      sensors,
      timeout: 60000, // can be slower for now as we request long db queries
    })) as StatValue[];
  }

  async notifyDevices() {
    return (await this.call({
      service: 'stats',
      method: 'notifyDevices',
    })) as string;
  }
}
